rotkehlchen/exchanges/bitfinex.py

```diff
@@ -802,6 +795,60 @@ def _process_unsuccessful_response(
         raise AssertionError(f'Unexpected {self.name} response_case: {case}')
 
     def first_connection(self) -> None:
+        """Request:
+        - currencies list: tickers of the supported assets.
+        - pairs list: pairs available for trading.
+        - currency map: a map between the ticker used by Bitfinex and the
+        standard one (outside Bitfinex).
+
+        These data are stored in properties along with `pair_bfx_symbols_map`,
+        a dict that maps a pair between the tickers of the base and quote assets.
+        """
+        if self.first_connection_made:
+            return
+
+        currencies_response = self._query_currencies()
+        if currencies_response.success is False:
+            raise RemoteError(
+                f'{self.name} failed to request currencies list. '
+                f'Response status code: {currencies_response.response.status_code}. '
+                f'Response text: {currencies_response.response.text}.',
+            )
+
+        exchange_pairs_response = self._query_exchange_pairs()
+        if exchange_pairs_response.success is False:
+            raise RemoteError(
+                f'{self.name} failed to request exchange pairs list. '
+                f'Response status code: {exchange_pairs_response.response.status_code}. '
+                f'Response text: {exchange_pairs_response.response.text}.',
+            )
+
+        currency_map_response = self._query_currency_map()
+        if currency_map_response.success is False:
+            raise RemoteError(
+                f'{self.name} failed to request exchange currency map. '
+                f'Response status code: {currency_map_response.response.status_code}. '
+                f'Response text: {currency_map_response.response.text}.',
+            )
+        # Generate a pair - tickers map. Currencies and pairs lists are already
+        # curated of test assets
```

Comment:
curated of test assets? I don't understand what this means.

Reply:
I meant the data has been curated (processed, including what we want, discarding what we don't want, etc). I've already modified the message.

Reply:
Thanks for the explanation. I now understand it.